File: AoC2022/solutions/06/solution.py
#Advent of Code 2022 day 6
from util import *
YEAR = 2022
DAY = 6


def get_data():
    return input_lines(DAY, YEAR)


data = get_data()[0]

def solve(window_size):
    for i in range(len(data)):
        if len(set(data[i:i+window_size])) == window_size:
            return i + window_size


first = solve(4)
second = solve(14)

print("First:  ", first)
print("Second: ", second)

File: AoC2022/solutions/06/solution.rs
//Advent of Code 2022 day 6
use std::fs::File;
use std::io::{BufRead, BufReader};

fn parse_input() -> Vec<String>{
    let filename = "AoC2022/solutions/06/input";
    let file = File::open(filename).unwrap();
    let reader = BufReader::new(file);
    let mut inputs: Vec<String> = Vec::new();

    for line in reader.lines(){
        inputs.push(line.unwrap());
    }

    return inputs;
}

pub fn solve() {
    let input = parse_input();
    
    let first  = 0;
    let second = 0;
    println!("First:  {}", first);
    println!("Second: {}", second);
}
